Controllers/userController.js

```diff
@@ -39,40 +39,57 @@ const authenticateToken = (req, res) => {
   });
 };
 
-const getAgePreference = async (req, res) =>{
+const getAgePreference = async (req, res) => {
   try {
-    const { minAge,maxAge } = req.body;
-    
-       const user = await User.findAll({
-        someAttribute: {
-          [Op.between]: [minAge, maxAge]
-         }})
-         if(user){
-          return res.status(201).send(user);
-         }
-        
-}catch (error) {
-  console.log(error);
-}
-}
+    const { email } = req.body;
+
+    // Find the user by email
+    const user = await User.findOne({
+      attributes: ['id', 'userName', 'email', 'age', 'minAge', 'maxAge'],
+      where: {
+        email: email,
+      },
+    });
+
+    if (user) {
+      const { minAge, maxAge } = user;
+
+      // Find all users with age within the specified range
+      const usersWithSameAgePreferences = await User.findAll({
+        attributes: ['id', 'userName', 'email', 'age', 'minAge', 'maxAge'],
+        where: {
+          age: {
+            [Op.between]: [minAge, maxAge],
+          },
+        },
+      });
+
+      return res.status(200).send(usersWithSameAgePreferences);
+    } else {
+      return res.status(404).send({ error: 'User not found' });
+    }
+
+  } catch (error) {
+    console.error(error);
+    return res.status(500).send({ error: 'Internal Server Error' });
+  }
+};
+
 const getUser = async (res) => { 
   
-  if(User){
-    return res.status(201)
-  }else {
-    return res.status(500)
-  }
+  console.log(User);
 }
 const signup = async (req, res) => {
  try {
-   const { userName, email, password, age, minAge, maxAge } = req.body;
+   const { userName, email, password, age, minAge, maxAge, images } = req.body;
```

Comment:
you could like just do
const data = {...req.body} 
data["password"] = await bcrypt.hash(password, 10)
To avoid listing all those variables twice